src/shims/foreign_items.rs

```diff
@@ -102,6 +104,36 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
     }
 
+    /// Lookup the body of a non-external foreign item.
+    fn lookup_non_external_foreign_item(
+        &self,
+        abi: Abi,
+        link_name: &str,
+    ) -> InterpResult<'tcx, Option<&'mir mir::Body<'tcx>>> {
+        let this = self.eval_context_ref();
+        let tcx = this.tcx.tcx;
+        let mut found = false;
+        let mut instance = None;
+        for cnum in iter::once(LOCAL_CRATE).chain(tcx.crates().iter().copied()) {
+            for &(symbol, _) in tcx.exported_symbols(cnum) {
+                if symbol_name_for_instance_in_crate(tcx, symbol, cnum) == link_name {
+                    if found {
+                        throw_ub_format!("symbol clashing detected: {}", link_name);
```

Comment:
This is likely not UB at this time, but I don't know how to emit a better error...

Reply:
`throw_unsup_format!` would be more appropriate then.